bitnami/external-dns/templates/dep-ds.yaml

```diff
@@ -587,7 +587,7 @@ spec:
               valueFrom:
                 secretKeyRef:
                   name: {{ template "external-dns.secretName" . }}
-                  key: rfc2136_tsig_secret
+                  key: {{ .Values.rfc2136.tsigKeyname }}
```

Comment:
why is this here? the secret.yaml template never set a key named the value of tsigKeyname.

Reply:
If the secret is created externally, not by the chart, and the key is named differently.

Reply:
this is what I meant: https://github.com/bitnami/charts/pull/17833

Reply:
this is what I meant: https://github.com/bitnami/charts/pull/18768/files#diff-ee55b5af91bd556b59c1dc234a5d17aae2f02e63287971530dc4b752a7b48ad0R593

Reply:
According to the description
>   ## rfc2136.secretName When using the rfc2136 provider, specify the existing secret which contains your tsig secret in the key "rfc2136_tsig_secret". Disables the usage of `rfc2136.tsigSecret` (optional)

"secretName" should be specified if the secret exist. This commit allows to mount such secret with a custom name of the key which holds the tsig secret.

Here is an example "values.yaml" for a cluster in which there is already a secret present. And this secret contains a the  tsig secret in a key not named "rfc2136_tsig_secret".

```yaml
# values.yaml
rfc2136:
  tsigKeyname: <custom-key-name-not-rfc2136_tsig_secret>
  secretName: <secret-created-earlier>
```
With such values.yaml the chart will not generate a new secret - there is no need.

Reverting this change will produce a deployment like this:
```yaml
# Source: external-dns/templates/dep-ds.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
spec:
  template:
    spec:
      containers:
        - name: external-dns
          args:
            - --rfc2136-tsig-keyname=<custom-key-name-not-rfc2136_tsig_secret>
          env:
            # RFC 2136 environment variables
            - name: EXTERNAL_DNS_RFC2136_TSIG_SECRET
              valueFrom:
                secretKeyRef:
                  name: <secret-created-earlier>
                  key: rfc2136_tsig_secret
```

Such deployment will not start because there is no key named "rfc2136_tsig_secret" inside secret "\<secret-created-earlier>".

I see no issue in this commit.

If you decide to specify a "rfc2136.tsigKeyname" in a combination with the tsig secret ("rfc2136.tsigSecret") - then you are creating an issue!
> ## rfc2136.tsigSecret When using the rfc2136 provider, specify the tsig secret to enable security. (do not specify if rfc2136.secretName is provided.) (optional)

Because the helm chart will generate a secret, when "rfc2136.tsigSecret" is specified, but will not use the name of the key in the deployment:

```yaml
# values.yaml
rfc2136:
  host: ipa-ca.iam.geek # replace with your host
  zone: noi.geek # replace with your zone
  tsigSecret: "<the-tsig-secret>"
  tsigKeyname: "<custom-key-name-not-rfc2136_tsig_secret>"
---
# Source: external-dns/templates/secret.yaml
apiVersion: v1
kind: Secret
metadata:
  name: testing-external-dns
type: Opaque
data:
  rfc2136_tsig_secret: "aXRJU0hFUkVFRUVF"
---
# Source: external-dns/templates/dep-ds.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
spec:
  template:
    spec:
      containers:
        - name: external-dns
          args: 
            - --rfc2136-tsig-keyname=<custom-key-name-not-rfc2136_tsig_secret>
          env:
            # RFC 2136 environment variables
            - name: EXTERNAL_DNS_RFC2136_TSIG_SECRET
              valueFrom:
                secretKeyRef:
                  name: testing-external-dns
                  key: rfc2136_tsig_secret
```

Reverting this commit will break "rfc2136.secretName". 

Is there any other case in which you will get "CreateContainerConfigError" because of such mismatch?